superset-frontend/src/explore/components/controls/CollectionControl/CollectionControl.test.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import CollectionControl from '.';
+
+jest.mock('@superset-ui/chart-controls', () => ({
+  InfoTooltipWithTrigger: (props: any) => (
+    <button
+      onClick={props.onClick}
+      type="button"
+      data-icon={props.icon}
+      data-tooltip={props.tooltip}
+    >
+      {props.label}
+    </button>
+  ),
+}));
+
+jest.mock('..', () => ({
+  __esModule: true,
+  default: {
+    TestControl: (props: any) => (
+      <button
+        type="button"
+        onClick={() => props.onChange(0, 'update')}
+        data-test="TestControl"
+      >
+        TestControl
+      </button>
+    ),
+  },
+}));
+
+let defaultProps = {
+  actions: {
+    addDangerToast: jest.fn(),
+    addInfoToast: jest.fn(),
+    addSuccessToast: jest.fn(),
+    addWarningToast: jest.fn(),
+    createNewSlice: jest.fn(),
+    fetchDatasourcesStarted: jest.fn(),
+    fetchDatasourcesSucceeded: jest.fn(),
+    fetchFaveStar: jest.fn(),
+    saveFaveStar: jest.fn(),
+    setControlValue: jest.fn(),
+    setDatasource: jest.fn(),
+    setDatasourceType: jest.fn(),
+    setDatasources: jest.fn(),
+    setExploreControls: jest.fn(),
+    sliceUpdated: jest.fn(),
+    toggleFaveStar: jest.fn(),
+    updateChartTitle: jest.fn(),
+  },
+  addTooltip: 'Add an item',
+  controlName: 'TestControl',
+  description: null,
+  hovered: false,
+  itemGenerator: jest.fn(),
+  keyAccessor: jest.fn(),
+  label: 'Time series columns',
+  name: 'column_collection',
+  onChange: jest.fn(),
+  placeholder: 'Empty collection',
+  type: 'CollectionControl',
+  validationErrors: [],
+  validators: [jest.fn()],
+  value: [{ key: 'hrYAZ5iBH' }],
+};
+
+beforeEach(() => {
+  defaultProps = {
+    actions: {
+      addDangerToast: jest.fn(),
+      addInfoToast: jest.fn(),
+      addSuccessToast: jest.fn(),
+      addWarningToast: jest.fn(),
+      createNewSlice: jest.fn(),
+      fetchDatasourcesStarted: jest.fn(),
+      fetchDatasourcesSucceeded: jest.fn(),
+      fetchFaveStar: jest.fn(),
+      saveFaveStar: jest.fn(),
+      setControlValue: jest.fn(),
+      setDatasource: jest.fn(),
+      setDatasourceType: jest.fn(),
+      setDatasources: jest.fn(),
+      setExploreControls: jest.fn(),
+      sliceUpdated: jest.fn(),
+      toggleFaveStar: jest.fn(),
+      updateChartTitle: jest.fn(),
+    },
+    addTooltip: 'Add an item',
+    controlName: 'TestControl',
+    description: null,
+    hovered: false,
+    itemGenerator: jest.fn(),
+    keyAccessor: jest.fn(),
+    label: 'Time series columns',
+    name: 'column_collection',
+    onChange: jest.fn(),
+    placeholder: 'Empty collection',
+    type: 'CollectionControl',
+    validationErrors: [],
+    validators: [jest.fn()],
+    value: [{ key: 'hrYAZ5iBH' }],
+  };
+});
+
+test('Should render', () => {
+  render(<CollectionControl {...defaultProps} />);
+  expect(screen.getByTestId('CollectionControl')).toBeInTheDocument();
```

Comment:
Just a tip, if you can't assign any reasonable role to the container component, you can always access it using:
 ```
const { container } = render(<CollectionControl {...defaultProps} />);
 expect(container).toBeInTheDocument();
```
😉

Reply:
I know, but I avoid using `const {container} = render()` as much as possible. I find it more reliable to use screen methods. I've had some problems with containers, so I avoid them whenever is possible.

Reply:
I'm curious what sort of problems you were running into there that might be worth avoiding them for. I like @michael-s-molina 's implementation - it's clean, and it makes less noise in the DOM.

Reply:
@rusackas It was not in this specific case. But in the past I've had problems that `container` had a very different result than `screen`. As `screen` is theoretically the HTML that will be rendered in the browser, I prefer to use it. Any change in behavior or configuration will not change the test result. That way I'm **sure** the test is right.

I only use `container` when I'm doing unit tests, and even in these cases I avoid it as much as possible.

Reply:
@yardz I don't know if in the past the implementation was different. Nowadays it's very similar. It will be different when we specify a `container` for `render` that is not the default `document.body`.

<img width="839" alt="Screen Shot 2021-03-26 at 11 32 38 AM" src="https://user-images.githubusercontent.com/70410625/112647174-fe394680-8e26-11eb-8cd5-3dd16aa65c7a.png">

<img width="833" alt="Screen Shot 2021-03-26 at 11 33 05 AM" src="https://user-images.githubusercontent.com/70410625/112647251-114c1680-8e27-11eb-9bd9-a3fbca90a371.png">

Kent C. Dodds in his excellent article [Common mistakes with React Testing Library](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen) recommends using `screen` whenever possible to avoid destructuring.

<img width="728" alt="Screen Shot 2021-03-26 at 11 34 01 AM" src="https://user-images.githubusercontent.com/70410625/112647367-33459900-8e27-11eb-94c2-0e6b178a3f24.png">

I also think we should always use `screen`. That being said, in the case where we CANNOT find a satisfactory role/text/placeholder for our top component and we just want to test that the `div` has rendered, both `test-id` or `container` are valid options. Since both options are not accessible, I prefer the one where we don't leave fingerprints in the component's code.

Reply:
Very nice post, several tips!

Well, I don't think that adding this test-id has any negative side effects at any level so the discussion comes down to "use or not the test-id." When it comes down to that it seems to me more like a discussion of "taste" than of "good practices".

If you think this is a block for the merge I can change it.

Reply:
@yardz For me, it's not a blocker as indicated by the ✔️.